astronomer/providers/dbt/cloud/triggers/dbt.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+import time
+from typing import Any, AsyncIterator, Dict, Optional, Tuple
+
+from airflow.providers.dbt.cloud.hooks.dbt import DbtCloudJobRunStatus
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+from astronomer.providers.dbt.cloud.hooks.dbt import DbtCloudHookAsync
+
+
+class DbtCloudRunJobTrigger(BaseTrigger):
+    """
+    DbtCloudRunJobTrigger is triggered with run id and account id, makes async Http call to dbt and get the status
+    for the submitted job with run id in polling interval of time.
+
+    :param conn_id: The connection identifier for connecting to Dbt.
+    :param run_id: The ID of a dbt Cloud job.
+    :param account_id: The ID of a dbt Cloud account.
+    :param poll_interval:  polling period in seconds to check for the status.
+    :param timeout: Time in seconds to wait for a job run to reach a terminal status for non-asynchronous
+        waits. Used only if ``wait_for_termination`` is True. Defaults to 7 days.
+    """
+
+    def __init__(
+        self,
+        conn_id: str,
+        poll_interval: float,
+        end_time: float,
+        run_id: int,
+        account_id: Optional[int],
+        wait_for_termination: bool = True,
+    ):
+        super().__init__()
+        self.run_id = run_id
+        self.account_id = account_id
+        self.conn_id = conn_id
+        self.end_time = end_time
+        self.poll_interval = poll_interval
+        self.wait_for_termination = wait_for_termination
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes DbtCloudRunJobTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.dbt.cloud.triggers.dbt.DbtCloudRunJobTrigger",
+            {
+                "run_id": self.run_id,
+                "account_id": self.account_id,
+                "conn_id": self.conn_id,
+                "end_time": self.end_time,
+                "poll_interval": self.poll_interval,
+                "wait_for_termination": self.wait_for_termination,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Make async connection to Dbt, polls for the pipeline run status"""
+        hook = DbtCloudHookAsync(self.conn_id)
+        try:
+            while await self.is_still_running():
+                if self.wait_for_termination and self.end_time < time.time():
```

Comment:
What if don't put `self.wait_for_termination` in the `if` condition here or don't use the variable at all in the trigger. How does it make a difference?

Reply:
async operator without the wait for completion does add much value. I agree with @pankajkoti and will be in favour of removing this. We can a docstring saying that it always wait for completion.  WDYT?

Reply:
ok, we can remove the `self.wait_for_termination` from the trigger and for async operator it will be always wait for completion